feature(server) User can signup

JackieBinya opened the pull request:
#### What does this PR do?
- adds endpoint user can sign up

#### Description of Task to be completed
- N/A

#### How should this be manually tested?
- Yet to implement unit testing.

#### Any background context you want to provide?
-  I implemented async-await, as advised on all middleware functions and I got the endpoint to finally work.
- No unit tests created as yet to test the endpoint, as I created a  new folder in 'server/src/v2' for my challenge 3, hence I'm expecting travis ci to fail. The previous tests were for challenge 2.


#### What are the relevant pivotal tracker stories?
[#167287420](https://www.pivotaltracker.com/n/projects/2356449)

#### Screenshots (if appropriate)
- N/A

JackieBinya committed:
feature(server) User can signup
- adds endpoint user can sign up
[finishes #167287420]

Mirriam-Maina commented on server/src/v2/middlewares/multer.js:
```diff
@@ -0,0 +1,38 @@
+      const mime = fileType(req.file.buffer);
+
+      // if can't be determined or format not accepted
+      if (!mime || !acceptedExtensions.includes(mime.ext)) { return next(new Error(`The uploaded file is not in  ${acceptedExtensions.join(',')} format!`)); }
```
Hey @JackieBinya you did not fix this return statements to use the response for the errors. As it is it will still crash the app

Mirriam-Maina commented on server/src/v2/middlewares/verify.js:
```diff
@@ -0,0 +1,82 @@
+import jwt from 'jsonwebtoken';
+import pool from '../models/configDB';
+
+export default class Middleware {
```
@JackieBinya rename this Middleware class to something more descriptive like AuthMiddleware since there are other middlewares in the application

Mirriam-Maina commented on server/src/v2/middlewares/verify.js:
```diff
@@ -0,0 +1,82 @@
+    next();
+  }
+
+  /* static async verifyExistingUser(req, res, next) {
```
@JackieBinya is this commented code is not being used; remove it. Otherwise, uncomment it and allow it to work with the current refactoring.

Mirriam-Maina reviewed (changes requested):
Kindly implement the changes

JackieBinya committed:
fature(server) Refactor user can signup
- reimplemented all middleware functions removed classes
- implemented all feedback from LFA
[finishes #167287420]
